JavaScript/JavaScriptSDK/Util.ts

```diff
@@ -220,8 +220,14 @@ module Microsoft.ApplicationInsights {
         /**
          * helper method to set userId and sessionId cookie
          */
-        public static setCookie(name, value) {
-            Util.document.cookie = name + "=" + value + ";path=/";
+        public static setCookie(name, value, domain?) {
+            var domainAttrib = "";
+
+            if (domain) {
+                domainAttrib = ";domain=" + domain;
```

Comment:
Not sure how it will be configurable, but we should allow do `if` logic by `location.host` so the same script will work in dev/test and production domain. And also will work when you have two domains like `abc.foo.com` and `cde.foo.ru`